x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/X-Block.java.st

```diff
@@ -58,6 +60,9 @@ $endif$
     @Override
     $Type$Block filter(int... positions);
 
+    @Override
+    ReleasableIterator<? extends $Type$Block> lookup(IntBlock positions, ByteSizeValue targetBlockSize);
```

Comment:
Should we make it as a default method, overrides subclasses where we can have optimizations?

Reply:
I thought about it and decided against it because:
1. I thought it'd be weird to stick it here - on `X-Block` instead of on `Block`. If it were on `Block` that'd be normal and good and easy. But putting it here feels a bit sneaky.
2. I think I'd remove it before *too* long anyway. I feel like I'm going to need a `Vector#lookup` before too long and that'd need the override here. I guess the others...  I dunno. I'm going to skip it for now.